src/Host/Client/Impl/Definitions/IRBlob.cs

```diff
@@ -0,0 +1,8 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.R.Host.Client {
+    public interface IRBlob {
```

Comment:
I think `IRBlobInfo` would be a more appropriate name (and `IRBlobData` could then become `IRBlob`). It would indicate that this is an object that _describes_ a blob, but doesn't actually contain one; whereas the actual blob can be both described and accessed. It would also make things more consistent where it meshes with files - e.g. right now you have `IRBlobFileInfo` derived from `IRBlob`, but with this change it'll be `IRBlobFileInfo : IRBlobInfo`.